frontend/src/data/game.ts

```diff
@@ -0,0 +1,204 @@
+import * as t from "io-ts";
+import {
+  API_ROOT,
+  GAME_LIST_CACHE_TIME,
+  GAME_LIST_STALE_TIME,
+} from "../global.config";
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+import { ParseJsonErrOptions, parseJsonResponse } from "./utils";
+
+export interface Game {
+  id: string;
+  name: string;
+  exchangeRate: number;
+  password: string;
+}
+
+// Custom error for public key query
+export class GameListQueryError extends Error {
+  constructor(message?: string) {
+    super(message);
+    this.name = "Game list query error";
+  }
+}
+
+export interface GameListVariables {
+  meta: {
+    jwt: string;
+  };
+}
+
+/**
+ * React hook for game list
+ */
+export const useGameList = (jwt: string) => {
+  return useQuery({
+    queryKey: ["gameList"],
+    cacheTime: GAME_LIST_CACHE_TIME,
+    staleTime: GAME_LIST_STALE_TIME,
+    queryFn: async (): Promise<Game[]> => {
+      // get current ticket from server
+      const res = await fetch(API_ROOT + "/games/", {
+        headers: {
+          Authorization: "Bearer " + jwt,
+        },
+      });
+
+      const data = await parseJsonResponse(
+        res,
+        t.array(
+          t.type({
+            id: t.union([t.number, t.string]),
+            name: t.string,
+            exchange_rate: t.number,
+            password: t.string,
+          })
+        ),
+        GameListQueryError
+      );
+
+      return data.map((item) => {
+        return {
+          id: typeof item.id === "number" ? item.id.toFixed() : item.id,
+          name: item.name,
+          exchangeRate: item.exchange_rate,
+          password: item.password,
+        };
+      });
+    },
+  });
+};
+
+// Custom error for coupon issue mutation
+export class CouponIssueError extends Error {
```

Comment:
Is this error used here? I think this was used a a template and you forgot to remove it.